src/test/c_api_tests.cpp

```diff
@@ -175,90 +219,110 @@ TEST(CApiConfigTest, MultiModelConfiguration) {
     ASSERT_NE(mmo, nullptr);
 
     // Cannot start due to configuration error
-    ASSERT_EQ(OVMS_ServerGeneralOptionsSetGrpcPort(go, 5555), nullptr);
-    ASSERT_EQ(OVMS_ServerGeneralOptionsSetRestPort(go, 5555), nullptr);  // The same port
-    ASSERT_EQ(OVMS_ServerMultiModelOptionsSetConfigPath(mmo, "/ovms/src/test/c_api/config.json"), nullptr);
+    ASSERT_CAPI_STATUS_NULL(OVMS_ServerGeneralOptionsSetGrpcPort(go, 5555));
+    ASSERT_CAPI_STATUS_NULL(OVMS_ServerGeneralOptionsSetRestPort(go, 5555));  // The same port
+    ASSERT_CAPI_STATUS_NULL(OVMS_ServerMultiModelOptionsSetConfigPath(mmo, "/ovms/src/test/c_api/config.json"));
 
     // Expect fail
-    // TODO: Check exact error code and details once error reporting becomes ready
-    ASSERT_NE(OVMS_ServerStartFromConfigurationFile(srv, go, mmo), nullptr);
+    ASSERT_CAPI_STATUS_NOT_NULL_EXPECT_CODE(OVMS_ServerStartFromConfigurationFile(srv, go, mmo),
+        StatusCode::OPTIONS_USAGE_ERROR);
 
     // Fix and expect ok
-    ASSERT_EQ(OVMS_ServerGeneralOptionsSetRestPort(go, 6666), nullptr);  // Different port
-    ASSERT_EQ(OVMS_ServerStartFromConfigurationFile(srv, go, mmo), nullptr);
+    ASSERT_CAPI_STATUS_NULL(OVMS_ServerGeneralOptionsSetRestPort(go, 6666));  // Different port
+    ASSERT_CAPI_STATUS_NULL(OVMS_ServerStartFromConfigurationFile(srv, go, mmo));
 
-    ASSERT_EQ(OVMS_ServerMultiModelOptionsDelete(mmo), nullptr);
-    ASSERT_EQ(OVMS_ServerGeneralOptionsDelete(go), nullptr);
-    ASSERT_EQ(OVMS_ServerDelete(srv), nullptr);
+    // Try to start again, expect failure
+    ASSERT_CAPI_STATUS_NOT_NULL_EXPECT_CODE(OVMS_ServerStartFromConfigurationFile(srv, go, mmo),
+        StatusCode::SERVER_ALREADY_STARTED);
+
+    // TODO: Is infer ok?
+
+    OVMS_ServerMultiModelOptionsDelete(mmo);
+    OVMS_ServerGeneralOptionsDelete(go);
+    OVMS_ServerDelete(srv);
+}
+
+TEST(CApiStatusTest, GetCodeAndDetails) {
+    std::unique_ptr<Status> s = std::make_unique<Status>(
+        StatusCode::INTERNAL_ERROR, "custom message");
+    OVMS_Status* sts = reinterpret_cast<OVMS_Status*>(s.get());
+    uint32_t code = 0;
+    ASSERT_CAPI_STATUS_NOT_NULL_EXPECT_CODE(OVMS_StatusGetCode(nullptr, &code), StatusCode::NONEXISTENT_STATUS);
+    ASSERT_CAPI_STATUS_NOT_NULL_EXPECT_CODE(OVMS_StatusGetCode(sts, nullptr), StatusCode::NONEXISTENT_STATUS_CODE_PTR);
+    ASSERT_CAPI_STATUS_NULL(OVMS_StatusGetCode(sts, &code));
+    const char* details = nullptr;
+    ASSERT_CAPI_STATUS_NOT_NULL_EXPECT_CODE(OVMS_StatusGetDetails(nullptr, &details), StatusCode::NONEXISTENT_STATUS);
+    ASSERT_CAPI_STATUS_NOT_NULL_EXPECT_CODE(OVMS_StatusGetDetails(sts, nullptr), StatusCode::NONEXISTENT_STATUS_DETAILS_PTR);
+    ASSERT_CAPI_STATUS_NULL(OVMS_StatusGetDetails(sts, &details));
+    std::stringstream ss;
+    ss << Status(StatusCode::INTERNAL_ERROR).string() << " - custom message";
+    EXPECT_EQ(std::string(details), ss.str());
+    OVMS_StatusDelete(reinterpret_cast<OVMS_Status*>(s.release()));
```

Comment:
Add test for trying to read Status(StatusCode::OK)

Reply:
done